File: 2024_1/Tecnologias Móveis/Aula03/atv02.kt
2 - Escreva uma função que recebe uma lista de números e retorna apenas os números pares.

fun numerosPares (lista: List<Int>): List<Int>{
    val numerosPares = mutableListOf<Int>()
    for(numero in lista) {
        if (numero % 2 == 0) {
            numerosPares.add(numero)
        }
    }
    return numerosPares;
}

fun main (){
    val listaOriginal = listOf (1, 3, 4, 20)
    val pares = numerosPares(listaOriginal)
    print("Numeros pares da lista original: $pares");
}

File: 2024_1/Desenvolvimento de Sistemas Móveis/Aula03/atv02.dart
// 2 - Escreva uma função que aceita uma lista de números como entrada e retorna uma nova lista onde cada elemento é o dobro do elemento correspondente na lista original.

import 'dart:io';
List<double> dobrarElementos(List<double> listaOriginal) {
    List<double> listaDobrada = [];

    for(double numero in listaOriginal){
        listaDobrada.add(numero * 2);
    }
    return listaDobrada;
}

void main (){
List<double> numeros = [2, 5, 10, 20];
List<double> numerosDobrados = dobrarElementos(numeros);
print('Lista original: $numeros');
print('Lista dobrados: $numerosDobrados');
}